src/Collection/Collection.router.ts

```diff
@@ -244,67 +247,63 @@ export class CollectionRouter extends Router {
   ): Promise<{ collection: FullCollection; items: FullItem[] }> => {
     const id = server.extractFromReq(req, 'id')
 
-    // We are using the withCollectionExists middleware so we can safely assert the collection exists
-    const [dbCollection, dbItems] = await Promise.all([
-      Collection.findOne<CollectionAttributes>(id),
-      Item.findOrderedByCollectionId(id),
-    ])
-    const remoteCollection = await collectionAPI.fetchCollection(
-      dbCollection!.contract_address!
-    )
-
-    if (!remoteCollection) {
-      // This might be a problem with the graph lagging but we delegate the retry logic on the client
-      throw new HTTPError(
-        'The collection is not published yet',
-        { id },
-        STATUS_CODES.unauthorized
-      )
-    }
-
-    const items: ItemAttributes[] = [...dbItems]
-    let remoteItems: ItemFragment[] = []
-
-    const isMissingBlockchainItemIds = dbItems.some(
-      (item) => item.blockchain_item_id == null
-    )
-
-    if (isMissingBlockchainItemIds) {
-      remoteItems = await collectionAPI.fetchItemsByContractAddress(
-        dbCollection!.contract_address!
-      )
+    try {
+      const dbCollection = await this.service.getDBCollection(id)
+
+      let result: { collection: CollectionAttributes; items: FullItem[] }
+
+      if (isTPCollection(dbCollection)) {
+        const itemIds = server.extractFromReq<string[]>(req, 'itemIds')
+        const dbItems = await Item.findByIds(itemIds)
+        result = await this.service.publishTPCollection(
+          dbCollection,
+          dbItems,
+          server.extractFromReq(req, 'signedMessage'),
+          server.extractFromReq(req, 'signature')
+        )
 
-      const updates = []
+        // Eventually, posting to the forum will be done from the server for both collection types (https://github.com/decentraland/builder/issues/1754)
+        // We should also consider deleteing Forum.router.ts
+        // DCL Collections posts are being handled by the front-end at the moment and the backend updated using '/collections/:id/post'
+        // TODO: Should this be halting the response? Retries?
+        // TODO: Consider slicing the items list if it might get too big
+        const forum_link = await createPost(
+          buildCollectionForumPost(result.collection, result.items)
+        )
```

Comment:
I think that the UI will keep the same retry mechanism for now, so we should be able to dismiss the publishing of the collection if there's no post created so if the post creation fails, the UI can retry it indefinitely.

Reply:
gotcha, I'll remove the `await` then